cmd/objectbox-gogen/gogen.go

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2020 ObjectBox Ltd. All rights reserved.
+ * https://objectbox.io
+ *
+ * This file is part of ObjectBox Generator.
+ *
+ * ObjectBox Generator is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * ObjectBox Generator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ObjectBox Generator.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// package main provides cmd line Main() function for objectbox-go project, actual command (executable) is part of objectbox-go project
```

Comment:
It's split because all the implementation in this project is internal (i.e. can't be used from other projects) so the integration is done this way to reduce the API surface just to the single function.

There's no technical reason to have two executables, there are practical one, however:  

See https://golang.objectbox.io/getting-started for how the generator is used by Go users
* comment `//go:generate go run github.com/objectbox/objectbox-go/cmd/objectbox-gogen` 
* due to the existing code that command must keep existing, at least for the time being
* actually, I'd say it should be kept this way because of simpler setup (no need to download objectbox-generator binary and add it to $PATH) as well as ensuring the right version of the generator is called (the one in user's go project's go.mod)

Reply:
BTW, I'll probably have a look at integrating this code with cmd/objectbox-generator, just for the sake of deduplicating the command code and having the option to generate Go code from the generator command, even though it's probably not so useful as I've described above.

Reply:
Trying to merge the objectbox-generator command with objectbox-gogen, I've stumbled upon a thing I've totally forgotten since working on this originally. Merging them into a single package isn't so simple because of the dependencies - we can't push c & flatc dependency onto the objectbox-gogen (i.e. objectbox-go package).

Therefore I've had to create a small interface that both executables need to implement with the things that differ among them. Also noticed OutPath is a shared option between both go and c generators so moved that to the shared options "on-the-fly"